heroic-elasticsearch-utils/src/main/java/com/spotify/heroic/elasticsearch/AbstractElasticsearchBackend.java

```diff
@@ -56,7 +54,7 @@ public void onFailure(Exception e) {
         Provider<T> emptyProvider, Runnable reportWriteDroppedByDuplicate
     ) {
         return throwable -> {
-            if (ExceptionUtils.getRootCause(throwable) instanceof VersionConflictEngineException) {
+            if (throwable.getMessage().contains("version conflict")) {
```

Comment:
Since we know what exceptions we want to deal with, why not just have a list of exceptions we act on, rather than string matching? I just wonder if the text will change at some point, which would cause this to break.

Reply:
Yeah that's what I was thinking as well. I think I tried going into the nested exception but I couldn't the other day.
Let me run it locally again and grab the exact errors and try again.